tools/qa/buildkite_static_code_analysis.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+if [ "$BUILDKITE_PULL_REQUEST" != "false" ]; then
+    source tools/qa/buildkite_common.sh
+    checkout_merge_commit
+    get_ancestor #writes $SHA variable
+
+    PATH=$PATH:~/.local/bin
+
+    echo "--- Running trapdoors tests"
+
+    TRAPDOORS="trapdoor_cppcheck.py
+    trapdoor_cpplint.py
+    trapdoor_doxygen.py
+    trapdoor_import.py
+    trapdoor_pycodestyle.py
+    trapdoor_pydocstyle.py"
+
+    for i in ${TRAPDOORS}; do
+        tools/qa/simulate_trapdoor_pr.py -vA $SHA tools/qa/$i
+    done
+fi
```

Comment:
So this is it: there is no newline at the end of the file. You don't have to add a empty line. You just need to make sure that the last line ends with a new-line character, which is the conventional thing to do. Try the following:

    toon@molmod55 ~/univ/code/horton1:buildkiteQA> cat tools/qa/buildkite_code_analysis.sh 
    #!/usr/bin/env bash

    if [ "$BUILDKITE_PULL_REQUEST" != "false" ]; then
        source tools/qa/buildkite_common.sh
        checkout_merge_commit
        get_ancestor #writes $SHA variable

        echo "--- Build refatoms"
        rm -rf data/refatoms/*.h5 data/refatoms/*.tar.bz2
        make -C data/refatoms/

        echo "--- Running trapdoors tests"
        ./tools/qa/check_whitespace.py || report_error "Whitespace errors in some commits"

        TRAPDOORS="trapdoor_coverage.py
        trapdoor_namespace.py"

        export PYTHONPATH=`pwd`
        export HORTONDATA=`pwd`/data

        for i in ${TRAPDOORS}; do
            tools/qa/simulate_trapdoor_pr.py -vrA $SHA tools/qa/$i
        done
    fitoon@molmod55 ~/univ/code/horton1:buildkiteQA> 

Note the `fi` being on the same line as my command line. It is unfortunate that `git diff --check` does allow for this.